g3doc/index.md

```diff
@@ -1072,6 +1072,19 @@ method on `NSArray`, which often comes handy, along with other similar
 [functional operators](https://github.com/google/functional-objc) that
 Objective-C lacks.
 
+
+Note: The number of promises `all` can handle is 4. If you need more than 4,
```

Comment:
Worth to mention that limitation appears only in Swift and only for promises of different types. Btw, for the [Any](text_features_size) operator the current limit is 3, may like to mention that in a corresponding docs section.

Reply:
Sure, it is worthy. I'll add it.  

I see that the swift-only and different promise types case has been mentioned in the previous paragraph.  

> **In Swift**, the all operator also allows passing promises of **heterogeneous types**. For this case, the returned promise will be resolved with a tuple containing the values of the input promises in the same order they were provided.

This paragraph looks great to put the limitation information.
And some lines of code seem helpful to give an example on how to solve this limitation issue. 
So it'll be added to where it was. 
Let me know if this is not fit.

Here is my checklist.
To add:

 - [ ] Swift Only
 - [ ] Applicable only to heterogeneous promises
 - [ ] Note for `any` limitation